Rakefile

```
@@ -32,12 +36,12 @@ rule ".rb" => ["%X.proto"] + codegen_rb_files do |t|
 
   unit = Tempfile.create(File.basename(t.source)) do |f|
     File.unlink(f.path)
-    sh("protoc -I #{File.dirname(t.source)} #{File.basename(t.source)} -o #{f.path}")
+    sh("protoc -I #{protoc_import_path_string} #{File.basename(t.source)} -o #{f.path}")
```

let's make this command an array so it isn't parsed by the shell, and then our args can be more straightforward

```suggestion
    sh("protoc", "-I", protoc_import_path_string, File.basename(t.source), "-o", f.path)
```

Also if you are changing the import path be sure that it works with the test proto files (which I believe this rule also covers)

---

> Also if you are changing the import path be sure that it works with the test proto files (which I believe this rule also covers)

And yes, a fair number of our tests fail if these proto files aren't generated properly.